[BunnyHoppp] iP

BunnyHoppp opened the pull request:
# Potong 
> "Potong red bean ~~🔴~~ is yummy 😋 "

## Features
Potong allows you to note down the things you need to do. It's,
- easy to use
- *text based*
- ** SUPER FUN **

### With Potong
1. You don't have to worry about your tasks
2. You can keep a list with you all the time
3. Do your grocery shopping 😄 

You can
- [x] Add tasks
- [ ] Edit tasks
- [ ] Combine tasks

Here is the [code](https://github.com/BunnyHoppp/ip.git) if you want to review.

If you are a Java programmer, you can use it to practice Java too. Here's the `main` method:
```java
public class Potong {
    public static void main(String[] args) {
        new Potong().run();
    }
}
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

BunnyHoppp committed:
Level-0: Rename, Greet, Exit

BunnyHoppp committed:
Level-1: Echo

BunnyHoppp committed:
Level-2: Add, List

BunnyHoppp committed:
Level-3: Mark as Done

BunnyHoppp committed:
Level-4: ToDos, Events, Deadlines

BunnyHoppp committed:
A-TextUiTesting

BunnyHoppp committed:
A-TextUiTesting

BunnyHoppp committed:
Level-5: Handle Errors

BunnyHoppp committed:
Level-6: Delete

BunnyHoppp committed:
Level-6: Delete

BunnyHoppp committed:
Level-6: Delete

BunnyHoppp committed:
Level-7: Save

BunnyHoppp committed:
Merge branch 'branch-Level-7'

BunnyHoppp committed:
Add Date and Time

BunnyHoppp committed:
Merge branch 'branch-level-8'

BunnyHoppp committed:
Add OOP for better structure

BunnyHoppp committed:
Merge branch 'branch-A-MoreOOP'

BunnyHoppp committed:
Add multiple packages

BunnyHoppp committed:
Merge branch 'branch-A-Packages'

BunnyHoppp committed:
Merge branch 'add-gradle-support'

BunnyHoppp committed:
Add JUnit tests

BunnyHoppp committed:
Merge branch 'branch-A-JUnit'

BunnyHoppp committed:
Change jar file name

BunnyHoppp committed:
Add find command

BunnyHoppp committed:
Add JavaDoc Comments

BunnyHoppp committed:
Change code to comply with coding standard

BunnyHoppp committed:
Add JavaDocs

BunnyHoppp committed:
Merge branch 'branch-A-JavaDoc'

BunnyHoppp committed:
Add JavaDocs

BunnyHoppp committed:
Add JavaDocs

AlphaPyke commented on src/main/java/potong/Parser.java:
```diff
@@ -0,0 +1,94 @@
+package potong;
+
+import potong.command.AddCommand;
+import potong.command.Command;
+import potong.command.DeleteCommand;
+import potong.command.ExitCommand;
+import potong.command.ListCommand;
+import potong.command.MarkCommand;
+import potong.command.FindCommand;
+
+import potong.exceptions.IllegalInputPotongException;
+
+import potong.task.DeadlineTask;
+import potong.task.EventTask;
+import potong.task.Task;
+import potong.task.ToDoTask;
+
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
```
I like the explicit imports.

AlphaPyke commented on src/main/java/potong/Parser.java:
```diff
@@ -0,0 +1,94 @@
+ */
+public class Parser {
+
+    private static final Pattern BASIC_COMMAND_FORMAT = Pattern.compile("(?<commandWord>\\S+)(?<arguments>.*)");
+
+    private static final Pattern DEADLINE_FORMAT = Pattern.compile(" (?<desc>.*)");
```
I like the screaming snake case used for the constants.

AlphaPyke commented on src/main/java/potong/command/AddCommand.java:
```diff
@@ -0,0 +1,118 @@
+    public AddCommand(String command, Type type) {
+        super(command);
+        this.type = type;
+        switch (type) {
+            case TODO -> this.task = command;
+            case DEADLINE -> {
+                String[] arr = command.split("/");
+                this.task = arr[0].strip();
+                this.deadline = arr[1].substring(3);
+            }
+            case EVENT -> {
+                String[] arr = command.split("/");
+                this.task = arr[0].strip();
+                this.eventStart = arr[1].substring(5).strip();
+                this.eventEnd = arr[2].substring(3).strip();
+            }
```
I believe switch-case should be indented as such?

AlphaPyke commented on src/main/java/potong/command/MarkCommand.java:
```diff
@@ -0,0 +1,46 @@
+ */
+public class MarkCommand extends Command {
+
+    private boolean toMark;
```
Should sound more like a boolean?
```suggestion
    private boolean isMarked;
```

AlphaPyke reviewed (commented):
LGTM. Minimal style/naming violations.

BunnyHoppp committed:
Add CheckStyle files

BunnyHoppp committed:
Merge branch 'branch-A-CheckStyle'

gavinsoh commented on src/main/java/potong/Parser.java:
```diff
@@ -0,0 +1,94 @@
+        final String commandWord = matcher.group("commandWord");
+        final String arguments = matcher.group("arguments").strip();
+
+        return switch (commandWord) {
+            case "bye" -> new ExitCommand(arguments);
+            case "list" -> new ListCommand(arguments);
+            case "mark" -> new MarkCommand(arguments, true);
+            case "unmark" -> new MarkCommand(arguments, false);
+            case "delete" -> new DeleteCommand(arguments);
+            case "todo" -> new AddCommand(arguments, AddCommand.Type.TODO);
+            case "deadline" -> new AddCommand(arguments, AddCommand.Type.DEADLINE);
+            case "event" -> new AddCommand(arguments, AddCommand.Type.EVENT);
+            case "find" -> new FindCommand(arguments);
+            default -> null;
```
I like how you've used the switch instruction to handle the different cases.

gavinsoh commented on src/main/java/potong/Storage.java:
```diff
@@ -0,0 +1,85 @@
+        this.tasklist.initialise(result);
+    }
+
+    /**
+     * Save the list into the file.
+     *
+     * @param textToAdd String representation of the list.
+     * @throws IOException If the input/output is wrong.
+     */
+    public void writeToFile(String textToAdd) throws IOException {
+        this.createFile();
+        FileWriter fw = new FileWriter(this.FILE_PATH);
+        fw.write(textToAdd);
+        fw.close();
+    }
```
I like how your methods are very clear in explaining their intended functionality.

gavinsoh commented on src/main/java/potong/TaskList.java:
```diff
@@ -0,0 +1,147 @@
+        }
+        return result.toString();
+    }
+}
```
I like how you have added JavaDocs comments for all your methods, it makes it very easy to understand your code.

BunnyHoppp committed:
Change some exception handling

BunnyHoppp committed:
Add GUI with dummy text

BunnyHoppp committed:
Fix GUI bugs

BunnyHoppp committed:
Add assertions

Add assertions for command classes so that assumptions about the program state is defined.

BunnyHoppp committed:
Tweak code quality

Change some structure in the code.

Still in progress.

BunnyHoppp committed:
Add more changes to code quality

BunnyHoppp committed:
Add stream processing for TaskList

Create a stream pipeline for easier list manipulation.

BunnyHoppp committed:
Merge pull request #1 from BunnyHoppp/branch-A-Assertions

Add assertions

BunnyHoppp committed:
Merge branch 'master' of https://github.com/BunnyHoppp/ip

BunnyHoppp committed:
Merge branch 'master' into branch-A-CodeQuality

BunnyHoppp committed:
Merge pull request #2 from BunnyHoppp/branch-A-CodeQuality

Improve code quality

BunnyHoppp committed:
Merge branch 'master' of https://github.com/BunnyHoppp/ip

BunnyHoppp committed:
Merge pull request #3 from BunnyHoppp/branch-A-Streams

Add stream processing for TaskList

BunnyHoppp committed:
Merge branch 'master' of https://github.com/BunnyHoppp/ip

BunnyHoppp committed:
Merge branch 'branch-A-CodeQuality' of https://github.com/BunnyHoppp/ip into branch-A-CodeQuality

BunnyHoppp committed:
Merge branch 'branch-A-CodeQuality'

BunnyHoppp committed:
Add tagging for tasks

Identify the tasks by tag.

Done in a same way as marking.

BunnyHoppp committed:
Fix some imports

BunnyHoppp committed:
Merge pull request #4 from BunnyHoppp/branch-C-Tagging

Add tagging for tasks

BunnyHoppp committed:
Merge branch 'master' of https://github.com/BunnyHoppp/ip

BunnyHoppp committed:
Fix some issue with storing data

Change Storage.createDirectory so that a directory can be created properly.

Jar file can now open properly

BunnyHoppp committed:
Add handling of wrong commands

Output the list of possible commands when user input a wrong command

Help user to know how to use Potong

BunnyHoppp committed:
Improve the GUI

Add some improvements to the GUI images and colour

Add icon and title to the GUI

Add exit by command

BunnyHoppp committed:
Add introduction to GUI

Welcome the users when first opening the app

BunnyHoppp committed:
Add README

Help users understand how to use Potong

Give detailed guide on the commands

BunnyHoppp committed:
Fix style issue

Run through all files to check the style

Keep code neat

BunnyHoppp committed:
Clean up exception handling

Check where to throw the exceptions and where to handle the exceptions

BunnyHoppp committed:
Improve exception handling

Allow user to know what went wrong with their input

Handle exception at the right place

BunnyHoppp committed:
Add more unit testing

Add more unit testing for TaskList and Task so to ensure that handling tasks is done well.

BunnyHoppp committed:
Improve code quality

Change some method names to be more descriptive so that it is clearer the anyone to read.

BunnyHoppp committed:
Fix some bugs that did not allow jar file to run

Bug for icon file name fixed. First letter is capitalised now

BunnyHoppp committed:
Add some example output to README

So that users know what to expect as an output.

BunnyHoppp committed:
Add error handling for wrong format for deadline

Wrong user input can be caught and handled

BunnyHoppp committed:
Add more description to README

Users know how tasks are displayed.
